### ISSUE TITLE
Xaml Hot Reload for Xamarin.Forms fails to start on Android devices in VS2019 (16.11.1)

### ISSUE THREAD

**Issue by DStilwagen** [viacheslav-lozinskyi/MetaOutput#7]
If you try to debug a Xamarin.Forms project deployed to an Android device or emulator the Xaml Hot Reload will fail to start with the following error. I initially commented on the bug reported here https://developercommunity.visualstudio.com/t/xaml-hot-reload-encountered-a-problem-and-failed-t-14/1503640 but I was able to solve the issue by disabling the MetaOutput extension. I'm not sure there is a problem with the extension it could very well be an issue with visual studio.

```
[4:41:33 PM]  XAML Hot Reload initializing...
[4:41:35 PM]  Caught exception in AgentStatusChangedHandler at 313: System.NullReferenceException: Failed to obtain Mono Debugger Evaluation Context.
   at Xamarin.HotReload.DebuggerExtensions.GetEvaluationContext(SoftDebuggerSession debugger) in D:\a\1\s\HotReload\Source\Xamarin.HotReload.Bridge\DebuggerExtensions.cs:line 33
   at Xamarin.HotReload.HotReloadBridge.<InitializeAgent>d__47.MoveNext() in D:\a\1\s\HotReload\Source\Xamarin.HotReload.Bridge\HotReloadBridge.cs:line 123
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Xamarin.HotReload.HotReloadBridge.<<Debugger_TargetHitBreakpoint>b__50_0>d.MoveNext() in D:\a\1\s\HotReload\Source\Xamarin.HotReload.Bridge\HotReloadBridge.cs:line 262
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.VisualStudio.Telemetry.WindowsErrorReporting.WatsonReport.GetClrWatsonExceptionInfo(Exception exceptionObject)
   at Xamarin.HotReload.DebuggerExtensions.GetEvaluationContext(SoftDebuggerSession debugger) in D:\a\1\s\HotReload\Source\Xamarin.HotReload.Bridge\DebuggerExtensions.cs:line 33
   at Xamarin.HotReload.HotReloadBridge.<InitializeAgent>d__47.MoveNext() in D:\a\1\s\HotReload\Source\Xamarin.HotReload.Bridge\HotReloadBridge.cs:line 123
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Xamarin.HotReload.HotReloadBridge.<<Debugger_TargetHitBreakpoint>b__50_0>d.MoveNext() in D:\a\1\s\HotReload\Source\Xamarin.HotReload.Bridge\HotReloadBridge.cs:line 262
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.VisualStudio.Telemetry.WindowsErrorReporting.WatsonReport.GetClrWatsonExceptionInfo(Exception exceptionObject)
[4:41:35 PM]  XAML Hot Reload encountered a problem and failed to start. Use Help -> Report a Problem if the issues persists.
[4:41:35 PM]  Stopping All Hot Reload sessions due to Failed ...
[4:41:35 PM]  (TestApp3.Android) Stopping a Hot Reload session due to Failed ...
[4:41:35 PM]  Stopped Hot Reload session.
[4:41:35 PM]  (TestApp3.Android) Caught exception in Debugger_TargetHitBreakpoint at 266: System.NullReferenceException: Failed to obtain Mono Debugger Evaluation Context.
   at Xamarin.HotReload.DebuggerExtensions.GetEvaluationContext(SoftDebuggerSession debugger) in D:\a\1\s\HotReload\Source\Xamarin.HotReload.Bridge\DebuggerExtensions.cs:line 33
   at Xamarin.HotReload.HotReloadBridge.<InitializeAgent>d__47.MoveNext() in D:\a\1\s\HotReload\Source\Xamarin.HotReload.Bridge\HotReloadBridge.cs:line 123
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Xamarin.HotReload.HotReloadBridge.<<Debugger_TargetHitBreakpoint>b__50_0>d.MoveNext() in D:\a\1\s\HotReload\Source\Xamarin.HotReload.Bridge\HotReloadBridge.cs:line 262
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.VisualStudio.Telemetry.WindowsErrorReporting.WatsonReport.GetClrWatsonExceptionInfo(Exception exceptionObject)
   at Xamarin.HotReload.DebuggerExtensions.GetEvaluationContext(SoftDebuggerSession debugger) in D:\a\1\s\HotReload\Source\Xamarin.HotReload.Bridge\DebuggerExtensions.cs:line 33
   at Xamarin.HotReload.HotReloadBridge.<InitializeAgent>d__47.MoveNext() in D:\a\1\s\HotReload\Source\Xamarin.HotReload.Bridge\HotReloadBridge.cs:line 123
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Xamarin.HotReload.HotReloadBridge.<<Debugger_TargetHitBreakpoint>b__50_0>d.MoveNext() in D:\a\1\s\HotReload\Source\Xamarin.HotReload.Bridge\HotReloadBridge.cs:line 262
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.VisualStudio.Telemetry.WindowsErrorReporting.WatsonReport.GetClrWatsonExceptionInfo(Exception exceptionObject)
```

**Comment by viacheslav-lozinskyi** [comment]
Thank you for reported bug.

Looking this log I don't see any function call related to MetaOutput, but in any case I will investigate the problem.
May be I need your addition assistance in reproducing of this bug.

**Comment by viacheslav-lozinskyi** [comment]
Could you clarify which version of MetaOutput, and which version of Android API you use?

**Comment by viacheslav-lozinskyi** [comment]
I successfully reproduced this bug.
It will be fixed in next release.

**Comment by viacheslav-lozinskyi** [comment]
Problem is solved in **MetaOutput 2.0.5**.
Please verify it.

I made workaround which detect this problem with "Hot Reload" initialization, and disable "Time Travel Debugging" in MetaOutput.
It is not a fix, because the problem not in MetaOutput, and there it just triggered.
Bug in Visual Studio (by log it is clearly visible).
